### PR TITLE
Bugfix/invalid commandl ine

### PR DESCRIPTION
**Pull request by tobyash86**
The PR fixes generation of invalid commandline when -project or -website are used, but there is also -solution added, because it has default value.

### CODE REVIEW

File: dist/sarif.xsl [hunk 1]
```diff
@@ -40,7 +40,9 @@
     
     <xsl:template name="rules_category">
         <xsl:param name="parentTags"/>
-        <xsl:variable name="tags" saxon:assignable="yes" select="concat($qt,@desc,$qt)"/>
+        <xsl:variable name="category_desc"><xsl:call-template name="escape_illegal_chars"><xsl:with-param name="text" select="@desc" /></xsl:call-template></xsl:variable>
+        <xsl:variable name="tags" saxon:assignable="yes" select="concat($qt,$category_desc,$qt)"/>
+
```
Comment (pkrukp) [inline dist/sarif.xsl:45]: Is this change related to invalid command line?

File: src/AnalysisRunner.ts [hunk 2]
```diff
@@ -60,7 +62,7 @@ export class AnalysisRunner
 
         // # Scope arguments
 
-        if(options.solution) {
+        if(options.solution && (!options.project && !options.website || options.solution != this.DEFAULT_SOLUTION_VALUE)) {
```
Comment (pkrukp) [inline src/AnalysisRunner.ts:65]: I'm not sure about this. If I understand correctly, if user specifies `solution: '.\\*.sln' ` and `project: '*.csproj'`, we will ignore solution file. But if user specifies `solution: '*.sln'` or `solution: 'some.sln'` plus a project, we will not ignore solution and command line will be still invalid.

Maybe we could either report somehow to user that command line is invalid, or ignore options which are invalid (with warning/console log?) but for all invalid values?

**Review by pkrukp — CHANGES_REQUESTED** [review]

File: dist/sarif.xsl (hunk 1 above)
Comment (tobyash86) [reply dist/sarif.xsl:45]: no, the xsl in dist folder was not up to date with the one in sources. I have fixed it

File: src/AnalysisRunner.ts (hunk 2 above)
Comment (tobyash86) [reply src/AnalysisRunner.ts:65]: I am not sure if we should decide for the user which variant of cmd is the desired one. User will already get failed workflow run, and in the details he will get information that the cmd was invalid. Also, he will have information that dotTEST exited with exit code specific to invalid commandline (we report each non-zero exit code, please see run-dottest-analyzer.ts).

Also, other tools does not have any validation. In fact user has to specify manually most of the switches for Jtest and CppTest.

Comment (pkrukp) [reply src/AnalysisRunner.ts:65]: It turns out to be problematic - github api will return the default value for the parameter if parameter is not explicitly specified by user. So it's not possible to detect that user has not specified any value for parameters with default value.

So I guess the current solution is good enough.

**Review by pkrukp — APPROVED** [review]
